src/task2/flattening_module/imageflattener.py

```diff
@@ -185,52 +221,64 @@ def done():
                 print('That\'s not a number.')
                 return
 
-            self.__dim = (short, long)
-
+            if self.__selected_panel is None:
+                print("No panel is selected")
+                pop.destroy()
+            self.__selected_panel.dims = (short, long)
+            self.__last_dims = (short, long)
             pop.destroy()
             self._flatten_img()
 
         done_button = Button(pop, text='Done', command=done)
         done_button.pack(side=LEFT)
 
-        if self.dim is not None:
+        if self.__last_dims is not None:
             skip_button = Button(pop, text='Use Last',
-                                 command=skip)
+                                 command=use_last)
             skip_button.pack(side=LEFT)
 
         pop.mainloop()
 
 
-def _parse_input(images_input: any) -> Dict[str, List[Tuple[int, int]]]:
+def parse_input(images_input: any) -> Dict[str, List[Panel]]:
     """Attempts to parse input from either list or dictionary if input is
     bad will throw useful errors"""
-
     panels_in = dict()
-
-    if isinstance(images_input, dict):
-        for key in images_input:
-            if isinstance(images_input[key], list):
-                panels_in[key] = []
-                for point in images_input[key]:
-                    if isinstance(point, list) \
-                       and len(images_input[key]) == 2 \
-                       and isinstance(point[0], int) \
-                       and isinstance(point[1], int):
-
-                        panels_in[key] = tuple(images_input[key])
-
-                    else:
-
-                        raise JsonFormatError('lists should contain lists of ints\
-                                              of len 2')
+    if not isinstance(images_input, list):
```

Comment:
Just for future reference, the conventional way to do this is via schema validation 
see : https://medium.com/grammofy/testing-your-python-api-app-with-json-schema-52677fe73351

Reply:
Thanks this is supper useful.